src/main/java/co/cask/gcp/speech/SpeechToTextTransform.java

```diff
@@ -0,0 +1,382 @@
+    @Description("Whether to mask profanity, replacing all but the initial character in each masked word with " +
+      "asterisks. For example, 'f***'.")
+    private String profanity;
+
+    @Macro
+    @Description("The language of the supplied audio as a [BCP-47](https://www.rfc-editor.org/rfc/bcp/bcp47.txt) " +
+      "language tag. Example: \"en-US\". See [Language Support](https://cloud.google.com/speech/docs/languages) for a" +
+      " list of the currently supported language codes.")
+    private String language;
+
+    @Macro
+    @Nullable
+    @Description("The name of the field to store all the different chunks of transcription with all the " +
+      "different possibility and their confidence score. Defaults to 'parts'")
+    private String transcriptionPartsField;
+
+    @Macro
+    @Nullable
+    @Description("If a field name is specified then the transcription with highest confidence score will be stored as" +
+      "text.")
+    private String transcriptionTextField;
+
+    @Name("serviceFilePath")
+    @Description("Path on the local file system of the service account key used for authorization. Can be set to " +
+      "'auto-detect' when running on a Dataproc cluster. When running on other clusters, the file must be present on " +
+      "every node in the cluster.")
+    @Macro
+    private String serviceAccountFilePath;
+
+    @Nullable
+    public String getServiceAccountFilePath() {
+      if (containsMacro("serviceFilePath") || serviceAccountFilePath == null ||
+        serviceAccountFilePath.isEmpty() || GCPConfig.AUTO_DETECT.equals(serviceAccountFilePath)) {
+        return null;
+      }
+      return serviceAccountFilePath;
+    }
+
+    @Nullable
+    public String getPartsField() {
+      if (containsMacro("transcriptionPartsField")) {
+        return null;
+      }
+      return transcriptionPartsField;
+    }
+
+    @Nullable
+    public String getTextField() {
+      if (containsMacro("transcriptionTextField")) {
+        return null;
+      }
+      return transcriptionTextField;
+    }
+
+    @Nullable
+    public String getAudioField() {
+      if (containsMacro("audiofield")) {
+        return null;
+      }
+      return audioField;
+    }
+
+    public boolean shouldFilterProfanity() {
+      return profanity.equalsIgnoreCase("true");
+    }
+
+    public String getLanguage() {
+      return language == null ? "en-US" : language;
+    }
+
+    @Nullable
+    public Integer getSampleRate() {
+      if (containsMacro("sampleRate")) {
+        return null;
+      }
+      return Integer.parseInt(sampleRate);
+    }
+
+    private void validate(Schema inputSchema) {
+      if (inputSchema != null) {
+        // If a inputSchema and audioField is available then verify that the schema does contain the given audioField
+        // and that the type is byte array. This will allow to fail fast i.e. during deployment time.
+        if (getAudioField() != null && inputSchema.getField(getAudioField()) == null) {
```

Comment:
create an audioFieldName variable to avoid repeated null checks and make it easier to read